src/test/java/htsjdk/samtools/cram/structure/ContainerTest.java

```diff
@@ -3,102 +3,83 @@
 import htsjdk.HtsjdkTest;
 import htsjdk.samtools.cram.CRAMException;
 import htsjdk.samtools.cram.ref.ReferenceContext;
-import htsjdk.samtools.cram.ref.ReferenceContextType;
-import org.testng.Assert;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import java.util.Arrays;
+import java.util.List;
 
 public class ContainerTest extends HtsjdkTest {
-    @Test
-    public static void singleRefSliceStateTest() {
-        final Slice slice = new Slice(new ReferenceContext(5));
-        slice.alignmentStart = 10;
-        slice.alignmentSpan = 15;
-
-        final Container container = Container.initializeFromSlices(Arrays.asList(slice));
-
-        Assert.assertTrue(container.getReferenceContext().isMappedSingleRef());
-        Assert.assertEquals(container.getReferenceContext().getType(), ReferenceContextType.SINGLE_REFERENCE_TYPE);
-        Assert.assertEquals(container.getReferenceContext(), slice.getReferenceContext());
-        Assert.assertEquals(container.alignmentStart, slice.alignmentStart);
-        Assert.assertEquals(container.alignmentSpan, slice.alignmentSpan);
+    @DataProvider(name = "containerStateTestCases")
+    private Object[][] containerStateTestCases() {
+        final ReferenceContext mappedReferenceContext = new ReferenceContext(5); // arbitrary
+
+        final int slice1AlignmentStart = 10;
+        final int slice1AlignmentSpan = 15;
+        final Slice slice1 = new Slice(mappedReferenceContext);
+        slice1.alignmentStart = slice1AlignmentStart;
+        slice1.alignmentSpan = slice1AlignmentSpan;
+
+        final int slice2AlignmentStart = 20;
+        final int slice2AlignmentSpan = 20;
+        final Slice slice2 = new Slice(mappedReferenceContext);
```

Comment:
Minor suggestion to improve the readability of the code below: use `mappedSlice` as a prefix for these variables.

Reply:
ok